src/resources/Search/SearchInterfaces.ts

```diff
@@ -39,6 +39,244 @@ export interface TokenModel {
     token: string;
 }
 
+export type RestQueryParams = PostSearchBodyQueryParams & PostSearchQueryStringParams;
+
 export interface PostSearchQueryStringParams {
-    viewAllContent?: boolean;
+    viewAllContent?: boolean | number;
+}
+
+export interface PostSearchBodyQueryParams {
+    actionsHistory?: RestActionHistory[];
+    analytics?: RestAnalyticsRequest;
+    aq?: string;
+    categoryFacets?: RestCategoryFacetRequest[];
+    childField?: string;
+    commerce?: RestCommerceParameters;
+    context?: RestContextRequest;
+    cq?: string;
+    debug?: boolean;
+    dictionaryFieldContext?: RestDictionaryFieldContextRequest;
+    dq?: string;
+    enableCollaborativeRating?: boolean;
+    enableDidYouMean?: boolean;
+    enableDuplicateFiltering?: boolean;
+    enableMLDidYouMean?: boolean;
+    enableQuerySyntax?: boolean;
+    excerptLength?: number;
+    facetOptions?: RestFacetOptions[];
+    facets?: RestFacetRequest[];
+    format?: RestFormat;
+    fieldsToExclude?: string[];
+    fieldsToInclude?: string[];
+    filterField?: string;
+    filterFieldRange?: number;
+    firstResult?: number;
+    groupBy?: RestGroupByRequest[];
+    index?: string;
+    indexToken?: string;
+    indexType?: string;
+    isGuestUser?: boolean;
+    locale?: string;
+    logicalIndex?: string;
+    lowercaseOperators?: boolean;
+    lq?: string;
+    lqPartialMatchKeywords?: string;
+    lqPartialMatchMaxKeywords?: number;
+    lqPartialMatchThreshold?: RestThreshold;
+    maximumAge?: number;
+    mlDidYouMeanMaxCandidates?: number;
+    mlDidYouMeanMinScore?: number;
+    mlDidYouMeanUseFacetCount?: boolean;
+    mlParameters?: RestMLParameters;
```

Comment:
This is actually a `Record<string,any>` on our end. So if you want to keep the type hints, maybe add an intersection with `any`

Reply:
So we are expecting undocumented values to be passed through this map? If so, I guess I'll simply type it as `Record<string, any>` then.

Reply:
I don't know what people are actually sending there but on SearchAPI, we don't do any specific parsing